skrf/time.py

```diff
@@ -287,47 +280,29 @@ def time_gate(ntwk, start: float = None, stop: float = None, center: float = Non
     window_width = abs(stop_idx - start_idx)
     window = signal.get_window(window, window_width)
 
+    nw_gated = ntwk.copy()
+
     # create the gate by padding the window with zeros
-    gate = npy.r_[npy.zeros(start_idx),
-                           window,
-                           npy.zeros(len(t) - stop_idx)]
-
-    #FFT the gate, so we have it's frequency response, aka kernel
-    kernel=fft.ifftshift(fft.fft(fft.fftshift(gate, axes=0), axis=0))
-    kernel =abs(kernel).flatten() # take mag and flatten
-    kernel=kernel/sum(kernel) # normalize kernel
-
-    out = ntwk.copy()
-
-    # conditionally delay ntwk, to center at t=0, this is
-    # equivalent to gating at center.  (this is probably very inefficient)
-    if center!=0:
-        out = out.delay(-center*1e9, 'ns',port=0,media=media)
-
-    # waste of code to handle convolve1d suck
-    re = out.s_re[:,0,0]
-    im = out.s_im[:,0,0]
-    s = convolve1d(re,kernel, mode=boundary)+\
-     1j*convolve1d(im,kernel, mode=boundary)
-    out.s[:,0,0] = s
-    # conditionally  un-delay ntwk
-    if center!=0:
-        out = out.delay(center*1e9, 'ns',port=0,media=media)
+    gate = npy.zeros_like(t)
+    gate[start_idx:stop_idx] = window
+
+    # time-domain gating
+    s_td = ifftshift(ifft(nw_gated.s[:, 0, 0]))
+    s_td_g = s_td * gate
+    nw_gated.s[:, 0, 0] = fft(fftshift(s_td_g))
 
     if mode == 'bandstop':
-        out = ntwk-out
+        nw_gated = ntwk - nw_gated
     elif mode=='bandpass':
         pass
     else:
         raise ValueError('mode should be \'bandpass\' or \'bandstop\'')
 
     if return_all:
-        # compute the gate ntwk and add delay
-        gate_ntwk = out.s11.copy()
-        gate_ntwk.s = kernel
-        gate_ntwk= gate_ntwk.delay(center*1e9, 'ns', media=media)
-
-        return {'gated_ntwk':out,
-                'gate':gate_ntwk}
+        # compute and return frequency response of time gate
+        nw_gate = nw_gated.copy()
+        nw_gate.s = ifftshift(fft(fftshift(gate)))
+        nw_gate.name = 'gate'
+        return nw_gated, nw_gate
```

Comment:
I just remembered I changed the output for `return_all=True` because I did not like the `Dict` format. Should I change it back? 
I'm wondering if this part could be removed completely. The `return_all` parameter has not even been documented previously, so not sure if anyone cares... Is there a use case where you would need the frequency-domain response of the time gate delivered in a 1-port Network?

```suggestion
        return {'gated_ntwk': nw_gated, 'gate': nw_gate}
```

Reply:
I would say to drop it as I do not like functions that does not return a fixed number of arguments (like in matlab), however I have no idea if this could be useful or not...

Reply:
I removed the parameter in 800966f, but now the builds of read-the-docs fail for some reason. The error output is very messy, so I don't know what's going on...

Reply:
@jhillairet Do you know how to debug this? I tried rerunning the checks, but it did not to rerun the build for read-the-docs. Building the docs locally works without a problem...